metronome/hotstuff/service/src/io/iohk/metronome/hotstuff/service/SyncService.scala

```diff
@@ -253,6 +254,25 @@ class SyncService[F[_]: Concurrent: ContextShift, N, A <: Agreement: Block](
         .void
   }
 
+  /** Validate the prepared block after the parent has been downloaded. */
+  private def validateBlock(block: A#Block): OptionT[F, Boolean] =
+    for {
+      parent <- OptionT {
+        storeRunner.runReadOnly {
+          blockStorage.get(Block[A].parentBlockHash(block))
+        }
+      }
+      // Reject the block straight away if it doesn't pass basic validation.
+      isFormallyValid =
+        Block[A].height(block) == Block[A].height(parent) + 1 &&
+          Block[A].isValid(block)
```

Comment:
There are 3 kinds of validations in this function:
- _consensus-specific syntactic validation_: height
- _application-specific syntactic validation_: content root
- _application-specific content validation_: interpreter-assisted

I'm worried that currently these look a bit like they're tied with a string. Perhaps we could capture these abstractions with better naming, but I realise that them being such simple checks this may look like an overkill.

Not a blocker for this PR, but I thought I'd point this out.

Reply:
Thanks, I also would have liked to see it broken up a bit, but didn't have any good ideas. Maybe I can move the the height check into an `isParentOf` method into the `Block`, so it doesn't need to be repeated. 

What exactly do you mean by "tied with a string"?

Reply:
Tried something, not sure it makes it much better.

Reply:
And another variant, maybe this one's good, there's no combination.